lib/rubocop/cop/lint/void.rb

```diff
@@ -143,8 +165,23 @@ def in_void_context?(node)
 
           return false unless parent && parent.children.last == node
 
+          return false if VALUE_RETURNING_METHODS.any? do |method|
+            enumerating_assigned?(parent, method) ||
+            enumerating_chained?(parent, method)
+          end
+
           VOID_CONTEXT_TYPES.include?(parent.type) && parent.void_context?
         end
+
+        def enumerating_chained?(parent, method)
+          method == parent.enumerated_method
+        end
+
+        def enumerating_assigned?(parent, method)
+          lvasgn_name, lvasgn_value = parent.parent&.lvasgn_return
+          lvasgn_value == method && !parent.enumerated.nil? &&
+            parent.enumerated == lvasgn_name
+        end
```

Comment:
I moved this two methods here from node.rb and removed #enumerating? 
https://github.com/rubocop-hq/rubocop/pull/7838#pullrequestreview-394320169
@bbatsov @koic